vitest.setup.js

```diff
@@ -0,0 +1,259 @@
+  /** @param {{ request: Request }} ctx */
+  return async ({ request }) => {
+    const url = new URL(request.url);
+    const d = url.searchParams.get('d');
+    if (!d) {
+      return;
+    }
+    const maybeJson = await commands.getCache([origin, route, d]);
+    if (maybeJson) {
+      return new HttpResponse(maybeJson);
+    }
+    const response = await fetch(bypass(request));
+    const json = await response.text();
+    await commands.setCache([origin, route, d], json);
+  };
+}
+
+/** @param {string} origin */
+function createHiGlassServerHandlers(origin) {
+  const tiles = createTilesOrTilesetInfoHandler(origin, 'tiles');
+  const tilesetInfo = createTilesOrTilesetInfoHandler(origin, 'tileset_info');
+  const chromsizes = createChromsizesHandler(origin, 'chrom-sizes');
+  const tilesets = createTilesetsHandler(origin, 'tilesets');
+  const suggest = createSuggestHandler(origin, 'suggest');
+  const viewconfs = createViewconfHandler(origin, 'viewconfs');
+  const availableChromsizes = createAvailableChromsizesHandler(
+    origin,
+    'available-chrom-sizes',
+  );
+  return [
+    // HTTP
+    http.get(`http://${origin}/tiles`, tiles),
+    http.get(`http://${origin}/tileset_info`, tilesetInfo),
+    http.get(`http://${origin}/chrom-sizes`, chromsizes),
+    http.get(`http://${origin}/tilesets`, tilesets),
+    http.get(`http://${origin}/suggest`, suggest),
+    http.get(`http://${origin}/viewconfs`, viewconfs),
+    http.get(`http://${origin}/available-chrom-sizes`, availableChromsizes),
+    // HTTPS
+    http.get(`https://${origin}/tiles`, tiles),
+    http.get(`https://${origin}/tileset_info`, tilesetInfo),
+    http.get(`https://${origin}/chrom-sizes`, chromsizes),
+    http.get(`https://${origin}/tilesets`, tilesets),
+    http.get(`https://${origin}/suggest`, suggest),
+    http.get(`https://${origin}/viewconfs`, viewconfs),
+    http.get(`https://${origin}/available-chrom-sizes`, availableChromsizes),
+  ];
+}
+
+if (import.meta.env.VITE_USE_MOCKS === '1') {
+  const worker = setupWorker(
+    ...createHiGlassServerHandlers('higlass.io/api/v1'),
+    ...createHiGlassServerHandlers('resgen.io/api/v1'),
+    ...createHiGlassServerHandlers('resgen.io/api/v1/gt/paper-data'),
+    http.get('http://s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv', () =>
+      HttpResponse.text(
+        `
+chr1	249250621
+chr2	243199373
+chr3	198022430
+chr4	191154276
+chr5	180915260
+chr6	171115067
+chr7	159138663
+chr8	146364022
+chr9	141213431
+chr10	135534747
+chr11	135006516
+chr12	133851895
+chr13	115169878
+chr14	107349540
+chr15	102531392
+chr16	90354753
+chr17	81195210
+chr18	78077248
+chr19	59128983
+chr20	63025520
+chr21	48129895
+chr22	51304566
+chrX	155270560
+chrY	59373566
+chrM	16571`.trim(),
+      ),
+    ),
+    http.get(
+      'https://s3.amazonaws.com/pkerp/public/gpsb/small.chrom.sizes',
+      () =>
+        HttpResponse.text(
+          `
+foo	249250621
+bar	243199373`.trim(),
+        ),
+    ),
```

Comment:
Seems like this would be a nice win" to also avoid relying on these static files.